### PR TITLE
FIPS: Add pom profile to build fips compliant boringSSL netty-tcnative

### PR DESCRIPTION
**Pull request by k-raina**
### Motivation:
As discussed in [issue](https://github.com/netty/netty-tcnative/issues/799), considering the growing demand for FIPS compliance in security-sensitive environments, an official netty-tcnative release supporting FIPS validation would greatly benefit the open-source community. This would simplify integration and provide a reliable, community-supported solution.

### Setup Configurations:
Tools: cmake 3.20, ninja build 1.10.0, clang-12, golang, java 11, maven 3.6.3, libapr1, automake, autoconf, libtool, libunwind-dev, pkg-config

Fips validated BoringSSL commit used is 853ca1ea1168dff08011e5d42d94609cc0ca2e27

### Build Steps: 

- Run Maven 
```
 mvn clean install -f boringssl-static/pom.xml -Pfips-boringssl-static
```

- While build is running you should see in logs:
```
...
Boringssl is fips compliant
...
```
- After build steps are completed you should see Jars eg.
```
.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.61.Final/netty-tcnative-boringssl-static-2.0.61.Final.jar
.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.61.Final/netty-tcnative-boringssl-static-2.0.61.Final-linux-x86_64.jar
```

### Modifications:
- Added pom profile `fips-boringssl-static` for fips compliant

### Tested on: 
Tested on linux AMD and ARM machine, which are supported as per FIPS security document attached in reference.
Output: https://drive.google.com/file/d/1eAFUIrHLbB7xiTpxHPs__N3Ha_Ltli76/view?usp=sharing

### Reference: 
Guidance on how to build FIPS validated modules: https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp4407.pdf

### CODE REVIEW

**Comment by normanmaurer** [comment]
I think we could pull this in to make things easier for people. I am not sure yet about including this in our release process. But thats a different discussion

**Comment by k-raina** [comment]
Thanks @normanmaurer ,
I have added sample output and build details in the description. 
Please let me know if anything else is needed to merge this branch  to master? Thanks in Advance

File: boringssl-static/pom.xml [hunk 1]
```diff
@@ -75,6 +75,282 @@
...
+              <execution>
+                <id>install-fips-boringssl</id>
+                <phase>process-sources</phase>
+                <goals>
+                  <goal>wget</goal>
+                </goals>
+              </execution>
+            </executions>
+            <configuration>
+              <url>https://commondatastorage.googleapis.com/chromium-boringssl-fips/boringssl-853ca1ea1168dff08011e5d42d94609cc0ca2e27.tar.xz</url>
```
Comment (normanmaurer) [inline boringssl-static/pom.xml:112]: How would we keep track of the right tar ball to download ?

Comment (k-raina) [reply boringssl-static/pom.xml]: Updated in commit: d1e16bac4954fa2ad2ec9b17143232e256e42350

**Comment by normanmaurer** [comment]
@k-raina can you please sign our icla: https://netty.io/s/icla and let me know once done ?

**Comment by k-raina** [comment]
@normanmaurer I have signed icla. Thanks

File: boringssl-static/pom.xml [hunk 2]
```diff
@@ -75,6 +75,282 @@
...
   <profiles>
+
+    <!-- The profile that builds a fips-boringssl-static jar -->
+    <profile>
+      <id>fips-boringssl-static</id>
+      <properties>
+        <boringsslCheckoutDir>${project.build.directory}/boringssl-${boringsslBranch}/boringssl</boringsslCheckoutDir>
+        <boringsslBuildDir>${boringsslCheckoutDir}/build</boringsslBuildDir>
+        <!--   Latest FIPS compliant boringSSL commit   -->
+        <boringsslBranch>${env.BORINGSSL_COMMIT}</boringsslBranch>
```
Comment (normanmaurer) [inline boringssl-static/pom.xml:86]: Why not just "hardcode" it the same as we do with the other boringssl build ?

Comment (k-raina) [reply boringssl-static/pom.xml:86]: @normanmaurer In sometime BooringSSL might release another FIPS validated build. That is why i feel we should not hardcode it. https://boringssl.googlesource.com/boringssl/+/master/crypto/fipsmodule/FIPS.md

Comment (k-raina) [reply boringssl-static/pom.xml]: @normanmaurer Thanks. Updated commit

**Comment by normanmaurer** [comment]
@k-raina thanks a lot!

**Pull request merged**